### ISSUE TITLE
Error when returning a parametrized List

### ISSUE THREAD

**Issue by jpprade** [mulesoft-labs/raml-for-jax-rs#369]
Hello
With 3.0.2 I was able to generate the raml file then I added this class :

```
@Path("/GetAircrafts")
public class AircraftRestService  extends GenericRestService {

	@Path("/")
	@Produces("application/json")
	@GET
	public List<AircraftRest> getAircrafts(@Context final UriInfo uriInfo,
			@Context final HttpServletRequest request) {
		List<AircraftRest> aircraftsRet = new ArrayList<AircraftRest>();
		.....
		return aircraftsRet; 
	}
}
```

Since then I am getting the error :

> [ERROR] Failed to execute goal org.raml.jaxrs:jaxrs-to-raml-maven-plugin:3.0.2:jaxrstoraml (default) on project basepub: A type incompatibility occurred while executing org.raml.jaxrs:jaxrs-to-raml-maven-plugin:3.0.2:jaxrstoraml: sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl cannot be cast to java.lang.Class

I guess it is because I return a List. When I change it to return an `Object `instead of a` List<AircraftRest>` it works

The full stack :

> Caused by: org.apache.maven.plugin.PluginExecutionException: A type incompatibility occurred while executing org.raml.jaxrs:jaxrs-to-raml-maven-plugin:3.0.2:jaxrstoraml: sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl cannot be cast to java.lang.Class
> at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:196)
>         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
>         ... 20 more
> Caused by: java.lang.ClassCastException: sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl cannot be cast to java.lang.Class
>         at org.raml.jaxrs.parser.source.RoasterSourceParser.getDocumentationFor(RoasterSourceParser.java:71)
>         at org.raml.jaxrs.parser.model.JerseyJaxRsEntity.getDescription(JerseyJaxRsEntity.java:49)
>         at org.raml.jaxrs.converter.model.JaxRsRamlEntity.getDescription(JaxRsRamlEntity.java:51)
>         at org.raml.jaxrs.types.TypeRegistry$1.describe(TypeRegistry.java:46)
>         at org.raml.jaxrs.types.RamlType$Fixer.adjustType(RamlType.java:230)
>         at org.raml.pojotoraml.RamlAdjuster$Composite.adjustType(RamlAdjuster.java:67)
>         at org.raml.pojotoraml.PojoToRamlImpl.handleSingleType(PojoToRamlImpl.java:80)
>         at org.raml.pojotoraml.PojoToRamlImpl.classToRaml(PojoToRamlImpl.java:40)
>         at org.raml.jaxrs.types.RamlType.write(RamlType.java:144)
>         at org.raml.jaxrs.types.TypeRegistry.writeAll(TypeRegistry.java:64)
>         at org.raml.emitter.ModelEmitter.emit(ModelEmitter.java:87)
>         at org.raml.emitter.FileEmitter.emit(FileEmitter.java:56)
>         at org.raml.jaxrs.raml.core.OneStopShop.parseJaxRsAndOutputRaml(OneStopShop.java:82)
>         at org.raml.jaxrs.plugin.JaxRsToRamlMojo.confinedExecute(JaxRsToRamlMojo.java:123)
>         at org.raml.jaxrs.plugin.JaxRsToRamlMojo.execute(JaxRsToRamlMojo.java:80)
>         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)

**Comment by jpbelang** [comment]
I fixed this in #359.  Can you try the snapshot ?

**Comment by jpprade** [comment]
I try but I can't make maven to find it, I added the repository serving it to my artifactory but it still can find the snapshot sorry
